CMakeLists.txt

```diff
@@ -1,19 +1,56 @@
-cmake_minimum_required(VERSION 3.5)
-
-#-----------------------------------------------------------------------------
-# Enable C++11
-#-----------------------------------------------------------------------------
-set(_msg "Setting C++ standard")
-message(STATUS "${_msg}")
-if(NOT DEFINED CMAKE_CXX_STANDARD)
-  if("${Slicer_VTK_VERSION_MAJOR}" STREQUAL "8" OR DEFINED Qt5_DIR)
-    set(CMAKE_CXX_STANDARD 11)
-  else()
-    set(CMAKE_CXX_STANDARD 98)
+# ==== Define cmake build policies that affect compilation and linkage default behaviors
+#
+# Set the SLICER_NEWEST_VALIDATED_POLICIES_VERSION string to the newest cmake version
+# policies that provide successful builds. By setting SLICER_NEWEST_VALIDATED_POLICIES_VERSION
+# to a value greater than the oldest policies, all policies between
+# SLICER_OLDEST_VALIDATED_POLICIES_VERSION and CMAKE_VERSION (used for this build)
+# are set to their NEW behaivor, thereby suppressing policy warnings related to policies
+# between the SLICER_OLDEST_VALIDATED_POLICIES_VERSION and CMAKE_VERSION.
+#
+# CMake versions greater than the SLICER_NEWEST_VALIDATED_POLICIES_VERSION policies will
+# continue to generate policy warnings "CMake Warning (dev)...Policy CMP0XXX is not set:"
+#
+set(SLICER_OLDEST_VALIDATED_POLICIES_VERSION "3.5")
+set(SLICER_NEWEST_VALIDATED_POLICIES_VERSION "3.13.1")
+cmake_minimum_required(VERSION ${SLICER_OLDEST_VALIDATED_POLICIES_VERSION} FATAL_ERROR)
+if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "${SLICER_NEWEST_VALIDATED_POLICIES_VERSION}")
+  #Set and use the newest available cmake policies that are validated to work
+  set(SLICER_CMAKE_POLICY_VERSION "${CMAKE_VERSION}")
+else()
+  set(SLICER_CMAKE_POLICY_VERSION "${SLICER_NEWEST_VALIDATED_POLICIES_VERSION}")
+endif()
+cmake_policy(VERSION ${SLICER_CMAKE_POLICY_VERSION})
```

Comment:
To avoid the complex logic, It may be sensible to require `3.13.1` after 4.10.1 is out

Reply:
CMake 3.13.2 was just announced! 😄

Reply:
:) Thanks.  My same argument holds for any cmake > 3.12

Reply:
By using CMake >= 3.12, we will be able to leverage the improved [cmake_policy](https://cmake.org/cmake/help/v3.12/command/cmake_policy.html) command and specify:

```
cmake_policy(VERSION 3.12...3.13.1)
```

Reply:
@jcfr . Any ETA on when this can be incorporated?  I'll be happy to address your comments and improve the simplified cmake_policy(VERSION 3.12...3.13.1)  as you suggest as a separate commit.